src/expr/src/scalar/func.rs

```diff
@@ -7467,10 +7467,18 @@ impl VariadicFunc {
         use VariadicFunc::*;
         let in_nullable = input_types.iter().any(|t| t.nullable);
         match self {
-            Coalesce | Greatest | Least => input_types
+            Greatest | Least => input_types
                 .into_iter()
                 .reduce(|l, r| l.union(&r).unwrap())
                 .unwrap(),
+            Coalesce => {
+                let nullable = !input_types.iter().any(|typ| !typ.nullable);
```

Comment:
Maybe it's more intuitive to write this as follows.
```
let nullable = input_types.iter().all(|typ| typ.nullable);
```
This changes the nullable property when there are no input types, which I don't know if it is valid for coalesce.

Reply:
> which I don't know if it is valid for coalesce

I hope it isn't, otherwise the `unwrap` would be incorrect!

Reply:
Calling `coalesce` with zero arguments is rejected, so all good on that front!

Reply:
Yes, the parser already rejects 0-arg `coalesce`.

I've changed the code to @antiguru's suggestion, thanks!